spec/ip_whitelist_checker_spec.rb

```diff
@@ -0,0 +1,233 @@
+        end
+      end
+
+      describe 'With ip "::1" and "::1"' do
+        it 'should return true' do
+          rule = {
+            'type' => 0,
+            'ip' => '::1',
+          }
+
+          is_matching = IpWhitelistChecker.is_ip_matches_rule('::1', rule)
+
+          expect(is_matching).to eq(true)
+        end
+      end
+
+      describe 'With ip "90.88.0.1" and "90.88.0.1"' do
+        it 'should return true' do
+          rule = {
+            'type' => 0,
+            'ip' => '90.88.0.1',
+          }
+
+          is_matching = IpWhitelistChecker.is_ip_matches_rule('90.88.0.1', rule)
+
+          expect(is_matching).to eq(true)
+        end
+      end
+
+      describe 'With ip "90.88.0.0" and "90.88.0.0"' do
+        it 'should return true' do
+          rule = {
+            'type' => 0,
+            'ip' => '90.88.0.0',
+          }
+
+          is_matching = IpWhitelistChecker.is_ip_matches_rule('90.88.0.0', rule)
+
+          expect(is_matching).to eq(true)
+        end
+      end
+
+      describe 'With ip "127.0.0.1" and "::1"' do
+        it 'should return true' do
+          rule = {
+            'type' => 0,
+            'ip' => '127.0.0.1',
+          }
+
+          is_matching = IpWhitelistChecker.is_ip_matches_rule('::1', rule)
+
+          expect(is_matching).to eq(true)
+        end
+      end
+
+      describe 'With range "90.88.0.0 - 90.88.255.255' do
+        it 'should return false' do
+          rule = {
+            'type' => 1,
+            'ip_minimum' => '90.88.0.0',
+            'ip_maximum' => '90.88.255.255',
+          }
+
+          is_matching = IpWhitelistChecker.is_ip_matches_rule('90.89.0.1', rule)
+
+          expect(is_matching).to eq(false)
+        end
+      end
+
+      describe 'With range "90.88.0.0 - 90.88.255.255' do
+        it 'should return false' do
+          rule = {
+            'type' => 1,
+            'ip_minimum' => '90.88.0.0',
+            'ip_maximum' => '90.88.255.255',
+          }
+
+          is_matching = IpWhitelistChecker.is_ip_matches_rule('::ffff:90.89.0.1', rule)
+
+          expect(is_matching).to eq(false)
+        end
+      end
+
+      describe 'With range "90.88.0.0 - 90.88.255.255' do
+        it 'should return true' do
+          rule = {
+            'type' => 1,
+            'ip_minimum' => '90.88.0.0',
+            'ip_maximum' => '90.88.255.255',
+          }
+
+          is_matching = IpWhitelistChecker.is_ip_matches_rule('90.88.118.79', rule)
+
+          expect(is_matching).to eq(true)
+        end
+      end
+
+      describe 'With range "90.88.0.1 - 90.88.255.255"' do
+        it 'should return true' do
+          rule = {
+            'type' => 1,
+            'ip_minimum' => '90.88.0.1',
+            'ip_maximum' => '90.88.255.255',
+          }
+
+          is_matching = IpWhitelistChecker.is_ip_matches_rule('90.88.118.79', rule)
+
+          expect(is_matching).to eq(true)
+        end
+      end
```

Comment:
For the 4 tests above a few context is missing, the rule is the same but not the value checked. It should be readable in the tests:

```ruby
describe 'With range rule "90.88.0.0 - 90.88.255.255' do
  rule = {
    'type' => 1,
    'ip_minimum' => '90.88.0.0',
    'ip_maximum' => '90.88.255.255',
  }

  describe 'check X value' do
  end
  
  describe 'check Y value' do
  end

  #... and so on
end
```